src/vcf-pdf-viewer.js

```diff
@@ -628,10 +627,13 @@ class PdfViewerElement extends
     }
 
     setCurrentPage(value) {
-        if (!this.__viewer || this.__loading) {
-            this.$.currentPage.value = value;
-        } else {
-            this.__pageChange();
+        if (value != undefined) {
+            if (!this.__viewer || this.__loading) {
```

Comment:
Wouldn't be better to take this outside the if clause, given that it is repeated?:
this.$.currentPage.value = value;

Reply:
Let me take a look

Reply:
I did some changes in the method as it was not working as expected